modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/stats/GeoIpDownloaderStats.java

```diff
@@ -44,6 +53,15 @@ public GeoIpDownloaderStats(StreamInput in) throws IOException {
         databasesCount = in.readVInt();
         skippedDownloads = in.readVInt();
         expiredDatabases = in.readVInt();
+        if (in.getTransportVersion().onOrAfter(GEOIP_ADDITIONAL_DATABASE_DOWNLOAD_STATS)) {
+            int databaseStatsCount = in.readVInt();
+            downloadedDatabaseInfos = new TreeMap<>();
+            for (int i = 0; i < databaseStatsCount; i++) {
+                downloadedDatabaseInfos.put(in.readString(), new DownloadedDatabaseInfo(in));
+            }
+        } else {
+            downloadedDatabaseInfos = null;
```

Comment:
Would it tidy the logic elsewhere very much if this were an empty map instead of null?

Reply:
I think that the reason I did that is to avoid sending "downloader_attempts" in the xcontent response to nodes that don't know anything about it. But nothing actually parses this class's xcontent other than unit tests, so maybe that's unnecessary. I'll remove this and make sure nothing breaks.